### ISSUE TITLE
Improve handling of git author email

### ISSUE THREAD

**Issue by bpg** [scala-steward-org/scala-steward-action#32]
Currently, the author name and email used by Scala Steward to make a PR are taken from a   GitHub user liked to the authentication token:

https://github.com/scala-steward-org/scala-steward-action/blob/b3852623ce5200c5a8eb3ca002407336b970fab2/src/github.ts#L12-L16

https://github.com/scala-steward-org/scala-steward-action/blob/b3852623ce5200c5a8eb3ca002407336b970fab2/src/main.ts#L35-L36

However, *if* the GitHub user has the [email privacy feature](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/blocking-command-line-pushes-that-expose-your-personal-email-address) enabled, the email address returned in the auth response is always `null`. As the result, commits made by Scala Steward don't have a valid identity, and also  can not be GPG-signed.

**Proposed improvements**:
- Explain in the documentation requirements to the user account referenced by the auth token: the full name and email address are going to be used to identify PRs made by Scala Steward.
- Add new action configuration parameter to specify overrides for author's email and username

**Comment by bpg** [comment]
@alejandrohdezma I'll make a PR if you're OK with the proposed changes.

**Comment by alejandrohdezma** [comment]
@bpg that sounds great! This will also allow synergies with the [crazy-max/ghaction-import-gpg](https://github.com/crazy-max/ghaction-import-gpg) Github Action since it returns the name/email of the GPG key:

```yaml
- name: Import GPG key
  id: import_gpg
  uses: crazy-max/ghaction-import-gpg@v2
  env:
    GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
    PASSPHRASE: ${{ secrets.PASSPHRASE }}
- name: Launch Scala Steward
  uses: scala-steward-org/scala-steward-action@v2
  with:
    github-token: ${{ secrets.ADMIN_GITHUB_TOKEN }}
    author-email: ${{ steps.import_gpg.outputs.email }}
    author-name: ${{ steps.import_gpg.outputs.name }}
```

**Comment by alejandrohdezma** [comment]
Fixed by #38! Thanks!
